src/main/kotlin/app/ui/AudioPlayer.kt

```diff
@@ -0,0 +1,10 @@
+import javafx.scene.media.AudioClip
+
+fun playKotlin(URI: String) {
+    val myClip = AudioClip(URI)
+    myClip.play()
+    while(myClip.isPlaying) {
+        Thread.yield()
```

Comment:
this is still a busy wait, and ultimately the player shouldn't be responsible for making this kind of thread management decision- AudioClip almost certainly spawns an internal thread to play on so what you're likely doing here is making decisions about the app's main thread.